src/Schema/Keywords/Maximum.php

```diff
@@ -39,9 +42,14 @@ class Maximum extends BaseKeyword
     public function validate($data, $maximum, bool $exclusiveMaximum = false): void
     {
         try {
-            Validator::numeric()->assert($data);
-            Validator::numeric()->assert($maximum);
-        } catch (ExceptionInterface $e) {
+            if (class_exists(NumericVal::class)) {
```

Comment:
Sorry for long delay. @t0mmy742 can you please try to implement a wrapper discussed here (but with a `method_exists` check instead of version check) ?

Or at least replace class check to the `method_exists` here (and other similar places) since we are not relying on checked class here but on the method in the `Validator` class

Reply:
Oh, I've read you note on PR body. Yeah, maybe implementing `DataValidator` is not worth here, but at least we should use `method_exists` instead of `class_exists` here.

Reply:
`method_exists` can not find this method since it is called with magic functions (`__call` and `__callStatic`). Using `is_callable` function should do the work, if you're ok with this 😅

Reply:
uff. yea,  I just don't want to rely on the check which result is not being used. I.e they can delete or rename this class next version but keep `numericVal` static method with some kind of workaround. check will fail but code actually keep working.
let's try `is_callable`. If it's not working - well we can keep this as is

Reply:
I can't modify right now, I will try after work, or next days

Reply:
@scaytrase I just tested with `is_callable`. It always returns true. In fact, we can call a method with every name, since it uses `__callStatic` function. 
About them renaming or deleting this class, it should not happen since it's part of their public API, and that would mean they would be creating a BC break (for minor version at least).
If you really don't like this method, you should know that calling `Validator::method()` (with `method` being an incorrect name) result in throwing a `\Respect\Validation\Exceptions\ComponentException` exception (v1 AND v2). We could do something like this:
```php
try {
    try {
        Validator::numericVal()->assert($data);
        Validator::numericVal()->positive()->assert($maximum);
    } catch (ComponentException$e) {
        Validator::numeric()->assert($data);
        Validator::numeric()->positive()->assert($maximum);
    }
} catch (Exception | ExceptionInterface $e) {
    throw InvalidSchema::becauseDefensiveSchemaValidationFailed($e);
}
```

It's up to you to decide what you prefer :)

Reply:
No the last one is perfomance-heavy, since you need to call a method, raise an exception and process it every time you have v1 installed. let stick to original plan then